pages/docs/tutorials/native/working-with-klib.md

```diff
@@ -0,0 +1,140 @@
+
+The concept of a library, a collection of resources to be used by applications, is available in Kotlin/Native and it is represented
+as a file with the extension `klib`. Kotlin/Native libraries are artifacts that are only available at compile time, i.e. we do not ship these as dependencies to our application
+(as we would for instance with DLL files on the Windows platform).
+
+A `klib` file is a compressed archive in zip format with the following directory structure:
+
+File `utils.klib`
+
+```
+    targets/                    /* One or more target platforms */
+    
+        macbook/                /* When targeting macbook, this entry would appear */
+        
+            kotlin/             /* Kotlin code compiled to LLVM bitcode */
+            
+            native/             /* Bitcode for additonal native objects */ 
+            
+        linux/                  /* When targeting Linux, this entry would appear */
+        
+            kotlin/
+            
+            native/
+            
+    linkdata/                   /* A set of [ProtoBuf](https://github.com/google/protobuf) files with serialised linkage metadata */ 
+    
+    resources/                  /* Resource files such as images, etc. */
+
+    manifest                    /* A file in Java Property Format describing the library */ 
+```
+
+**Note**: It's important to note that until Kotlin/Native stabilises, the internal format of a `klib` file may change somewhat. Usually when working with `klib` files however,  
+knowing the internals is not required. 
+ 
+## Creating a Kotlin/Native Libraries
+
+There are two ways we can create a Kotlin/Native library. The first and most common way is to use the Kotlin compiler. Let's assume
+we create a small library called `utils.kt` which contains the following two functions
+
+```kotlin
+package utils
+
+fun printMessage(message: String) {
+    println("Message: $message")
+}
+
+fun printWarning(warning: String) {
+    println("Warning: $warning")
+}
+```
+
+To create a library from this file, we can use the compiler with the parameter `-produce library` or `-p library` for short:
+
+    konanc utils.kt -p library
+    
+By default, the output of the filename is `library.klib`. We can override it using the `-output` or `-o` parameter:
+
+    konanc utils.kt -p library -o utils
+    
+
+The second way to create a library is using the `cinterop` tool which allows us to create a Kotlin/Native library of an existing
+C library. See the [Interop with C tutorial](interop-with-c.html) on how to accomplish this.
+ 
+## Consuming a Kotlin/Native Library
+
+Now that we have our library, we can use it in our application. In our case this is a simple file named `sample.kt` with the following contents:
+
+```kotlin
+import utils.*
+
+fun main(args: Array<String>) {
+    printWarning("App is about to shut down!")
+}
+```
+
+Notice how we need to import the library on the first line using the `import` statement. 
```

Comment:
I think it's important to use precise wording here. The "import" statement does not import a library; it imports a package, and the declarations in this package are part of that library.